coding/value_opt_string.hpp

```diff
@@ -55,7 +48,7 @@ class StringNumericOptimal
   void Write(Sink & sink) const
   {
     uint64_t n;
-    if (ToInt(n))
+    if (ToInt(n) && (m_s.size() == 1 || m_s.front() != '0'))
```

Comment:
10K 0 values in osm https://taginfo.openstreetmap.org/tags/addr%3Ahousenumber=0#overview

Reply:
Yes, and it will be treated as 0-number (needs 1 byte for encoding). Unlike "01" will be treated as a string value (needs 3 bytes for encoding).